### PR TITLE
feat: Adds `Slideover` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

-  Moves `Slideover` component to @faststore/components
-  Adds `Slideover` styles on @faststore/ui
-  Review `Slideover` structure
- Creates `SlideOverHeader` component
- Replaces `SliderOver` occurrences
- Renames `HeroHeading` to `HeroHeader`

*** Doc page: [SlideOver doc page](https://faststore-site-git-feat-slideover-fs-768-faststore.vercel.app/components/organisms/slide-over)

## How to test it?

`run yarn dev` in core package and checks the SlideOver component in the components below:

- CartSideBar
- NavbarSlider
- FilterSlider

### Starters Deploy Preview

WIP

### CODE REVIEW

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  SlideOverProps,
+  useFadeEffect,
+  useUI,
+} from '@faststore/ui'
+
+export type SlideOverUsageProps = {
+  isOpen: SlideOverProps['isOpen']
+  size: SlideOverProps['size']
+  direction: SlideOverProps['direction']
+}
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:13]: We can use the [utility type Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) to simplify this type: 

```suggestion
export type SlideOverUsageProps = Pick<SlideOverProps, "isOpen" | "size" | "direction">
```

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+const SlideOverUsage = (props: SlideOverUsageProps) => {
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:15]: Please, use the destructuring in these props to follow the pattern

```suggestion
const SlideOverUsage = ({ isOpen, size, direction }: SlideOverUsageProps) => {
```

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const { closeModal } = useUI()
+  const { fade, fadeOut } = useFadeEffect()
+  return (
+    <SlideOver
+      fade={fade}
+      isOpen={props.isOpen}
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:21]:
```suggestion
      isOpen={isOpen}
```

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      onDismiss={() => fadeOut()}
+      size={props.size}
+      direction={props.direction}
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:24]:
```suggestion
      direction={direction}
```

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          alignItems: 'center',
+          justifyContent: 'center',
+          padding: '1rem',
+          height: '100%',
+        }}
+      >
+        <p>
+          A generic <code>SlideOver</code> component.
+          <br></br>
+          size:<code>{props.size}</code>
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:42]:
```suggestion
          size:<code>{size}</code>
```

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+          <br></br>
+          direction:<code>{props.direction}</code>
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:44]:
```suggestion
          direction:<code>{direction}</code>
```

File: apps/site/pages/components/organisms/hero.mdx [hunk 7]
```diff
@@ -27,7 +27,7 @@ The `Hero` component is a full-width banner presented on the above-the-fold sect
 The `Hero` component is a compound of the following:
 
 - `HeroImage`: wraps the Hero image.
-- `HeroHeading`: wraps the Hero textual content. It may contain a title, description, and a call-to-action button.
+- `HeroHeader`: wraps the Hero textual content. It may contain a title, description, and a call-to-action button.
```
Comment (ArthurTriis1) [inline apps/site/pages/components/organisms/hero.mdx:30]: Good renaming!

File: apps/site/pages/components/organisms/slide-over.mdx [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+    <>
+      <Button variant="primary" onClick={openModal} style={{ width: '9rem' }}>
+        Right
+      </Button>
+      {modal && (
+        <SlideOverUsage size="partial" direction="rightSide" isOpen={modal} />
+      )}
+    </>
+  )
+}
```
Comment (ArthurTriis1) [inline apps/site/pages/components/organisms/slide-over.mdx:63]: We could add an example of this code in the documentation.

File: apps/site/components/SlideOver/SlideOverUsage.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+const SlideOverUsage = ({ isOpen, size, direction }: SlideOverUsageProps) => {
+  const { closeModal } = useUI()
+  const { fade, fadeOut } = useFadeEffect()
+  return (
+    <SlideOver
+      fade={fade}
+      isOpen={isOpen}
+      onDismiss={() => fadeOut()}
+      size={props.size}
```
Comment (ArthurTriis1) [inline apps/site/components/SlideOver/SlideOverUsage.tsx:19]:
```suggestion
      size={props.size}
```

**Review by ArthurTriis1 — APPROVED** [review]

File: apps/site/pages/components/organisms/slide-over.mdx [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+<header>
+
+# SlideOver
+
+This component represents an aditional session that complements the screen's information.
+
+</header>
+
+---
+
```
Comment (renatamottam) [inline apps/site/pages/components/organisms/slide-over.mdx:34]:
```suggestion

```

File: apps/site/pages/components/organisms/slide-over.mdx [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+  />
+  <TokenRow
+    token="--fs-slide-over-partial-max-width-notebook"
+    value="calc(var(--fs-grid-breakpoint-notebook) / 3)"
+  />
+</TokenTable>
+
+---
+
+## Customization
```
Comment (renatamottam) [inline apps/site/pages/components/organisms/slide-over.mdx:271]: Fan, what about adding a "Related components" ([button example](https://evergreen.faststore.dev/components/atoms/button#related-components)) sections with `coming soon` links?

**Review by renatamottam — APPROVED** [review]
Great work!

File: packages/core/src/components/cart/CartSidebar/cart-sidebar.module.scss [hunk 12]
```diff
@@ -60,6 +44,8 @@
   }
 
   [data-fs-cart-sidebar-footer] {
+    position: sticky;
+    bottom: 0;
```
Comment (eduardoformiga) [inline packages/core/src/components/cart/CartSidebar/cart-sidebar.module.scss:48]: hmm, I wonder if this can make the footer is hidden in a real device. 👀

**Review by eduardoformiga — APPROVED** [review]
Good job, Fan!

File: apps/site/pages/components/organisms/slide-over.mdx [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+          {modal && (
+            <SlideOverUsage size="full" direction="rightSide" isOpen={modal} />
+          )}
+        </>
+      )
+    }
+    ```
+
+    ```tsx
+    export const SlideOverFullRight = () => {
```
Comment (eduardoformiga) [inline apps/site/pages/components/organisms/slide-over.mdx:183]:
```suggestion
    export const SlideOverFullLeft = () => {
```

File: apps/site/pages/components/organisms/slide-over.mdx [hunk 14]
```diff
@@ -0,0 +1,281 @@
+---
+description: This component represents an aditional session that complements the screen's information.
+keywords:
+  - UI library
+  - ecommerce
+  - react
+sidebar_custom_props:
+  image: https://vtexhelp.vtexassets.com/assets/docs/src/PaymentMethods___b0f344b254360daa53406cd631887adf.png
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply apps/site/pages/components/organisms/slide-over.mdx:8]: actually no, we need to review all this images later, or just remove it.

File: packages/core/src/components/cart/CartSidebar/cart-sidebar.module.scss (hunk 12 above)
Comment (hellofanny) [reply packages/core/src/components/cart/CartSidebar/cart-sidebar.module.scss:48]: I'll test it again when available on the starter. 🙇

File: apps/site/pages/components/organisms/slide-over.mdx (hunk 11 above)
Comment (hellofanny) [reply apps/site/pages/components/organisms/slide-over.mdx:268]: done [here](https://github.com/vtex/faststore/pull/1649/commits/9a1194d6b7460d65d6a680c5388eed87a0d7b362) :)

**Pull request merged**